### PR TITLE
Add --overwrite and --group flags

### PR DESCRIPTION
**Pull request by otherguy**
The `--overwrite` flag always creates the system service files, regardless of whether they exist.
The `--group` flag adds a `Group=...` line to the systemd service file.

*Other changes:*

* Fixed several typos and grammar mistakes
* Only display the `sudoers` hint if `VERBOSE` is enabled

### CODE REVIEW

**Comment by vifreefly** [comment]
Hello @otherguy , thanks for PR!

 Like you already noticed, in this project there are no mixed quotes, all quotes is double quotes. That means I prefer double quotes everywhere. There is nothing wrong with double quotes. It's a matter of choice.

Because of your changes (" to ') it's hard to see what also was changed. Same for https://github.com/vifreefly/capistrano-procsd/

**Comment by otherguy** [comment]
@vifreefly again, I'm sorry. You are absolutely right that it's a matter of choice which quotes you use.  Same as for https://github.com/vifreefly/capistrano-procsd/pull/1 I applied our company's Rubocop template without thinking. Apologies!

I have reverted the change. This should make it much easier to see my changes.

File: lib/procsd/cli.rb [hunk 1]
```diff
@@ -8,18 +8,20 @@ class ArgumentError < StandardError; end
 
     desc "create", "Create and enable app services"
     option :user, aliases: :u, type: :string, banner: "$USER", default: ENV["USER"]
+    option :group, aliases: :g, type: :string, banner: "$GROUP", default: ""
```
[the remark this answers is not in the document]
Comment (otherguy) [reply lib/procsd/cli.rb]: Updated to `nil`. I was sure `option` with `type: string` can only have ster

File: lib/procsd/templates/service.erb [hunk 2]
```diff
@@ -8,6 +8,9 @@ ReloadPropagatedFrom=<%= config["target_name"] %>
 [Service]
 Type=simple
 User=<%= config["user"] %>
+<% unless config["group"].empty? -%>
```
[the remark this answers is not in the document]
Comment (otherguy) [reply lib/procsd/templates/service.erb]: Updated to `nil`.

File: lib/procsd/cli.rb [hunk 3]
```diff
@@ -227,10 +228,8 @@ def __print_version
 
     def perform_create
       options.each do |key, value|
-        next unless %w(user dir path).include? key
-        if value.nil? || value.empty?
-          say("Can't fetch value for --#{key}, please provide it's as an argument", :red) and return
```
[the remark this answers is not in the document]
Comment (otherguy) [reply lib/procsd/cli.rb]: I agree with your design decision not to require `sudo` (hence my changes to `capistrano-procsd`)!

[the remark this answers is not in the document]
Comment (otherguy) [reply lib/procsd/cli.rb]: Added it back in 🙂

**Comment by vifreefly** [comment]
> How do you deal with Capistrano deployments then? Currently, any subsequent deploy will fail

Yes, if I updated Procfile, for now I have to manually `destroy` the services first, before doing `cap production deploy`. This is actually not good, because 1. I have to type another command `cap production procsd:destroy` 2. This approach makes application down (not working) for a some amount of time. 

> Currently `create --overwrite --or-restart` does both: (re-)creates the service definition, then restarts.

I would like to have option like **`--or-update-and-restart`** instead of combined `--overwrite` + `--or-restart`.

It has more predictable name, and does exactly what it does, without any confusion (and when `.procsd.lock` will be added, this option will be upgraded to perform a full update, not just overwriting  services like now). 
We still keep `or-restart`  for now, but both commands can not be combined. Only `procsd create --or-restart` or only `procsd create --or-update-and-restart`.

What do you think about it?

**Comment by otherguy** [comment]
How about we change the signatures completely?

* `create` creates the service definition. Optional `--or-update` flag updates it
* `update` updates the service definition (`destroy`, then `create`)
* `destroy` destroys it

None of these allow any restart/start/stop actions.

* `enable` and `disable` stay the same, as well `start`, `stop` and `restart`.

In your `deploy.rb` you can then do:

```
# deploy.rb

after  :finishing,   'procsd:update'
after  :finished,    'procsd:restart'
```

It's 2 commands but that should not be an issue. And `procsd:update` should destroy and re-create the service definitions without actually stopping the services.

Users who are constantly adding/removing services from their `Procfile`, could do a `procsd:stop` before doing `procsd:update`, at least until the `Procfile.lock` system is implemented.

That does away with `--or-restart` and the awkwardly named `--or-update-and-restart` completely.

**Comment by vifreefly** [comment]
> None of these perform any restart/start/stop actions.

Well, I think they should. At least `$ procsd destroy` should keep existing behavior and stop the services before deleting them if they are running. This is what this command should do - completely get rid of the services.

Also to analogy with `destroy`, the `create` command should by default start the services after creation (currently it doesn't start them automatically). 

And finally `procsd update` (which is not exist yet) should automatically restart the services after updating. So all 3 commands will have their appreciate default callbacks: create > start, destroy > stop, update > restart. Makes sense as for me.

> after :finishing, 'procsd:update'

My thoughts about `update` command:

1) Now settings like --dir, --user, --path (and --group) provided as (optional) arguments for a `create` command. If they are were provided with create command, we have to provide them for update command as well. That means duplication of cli options for both commands (create and update). I think it's better then to store these options in `procsd.yml` instead, and do not use cli arguments. Something like:

```yml
# procsd.yml

app: my_app
options:
  user: deploy
  group: deploy
  dir: /home/deploy/my_app/current
  path: /some/custom/path:$PATH
```
That gives the possibility to use same options for `create` and `update` commands and do not use cli arguments at all. Of course these settings stays optional, and by default any not provided options will be fetched from current environment (like now).

2.
> And `procsd:update` should destroy and re-create the service definitions without actually stopping the services.

I think it's fine for `update` to just rewrite services files + restart the target (because it's default update callback, see above). It will be even better to do nothing, and print something like `Nothing to update, Procfile and procsd.yml didn't changed. Restarting services only.` if there wasn't any changes in Procfile or procsd.yml. But it will be possible only with `.procsd.lock`. For now it's fine to  just rewrite files. 

3. 
>     * `create` creates the service definition. Optional `--or-update` flag updates it

I thinking about changing things, and use new logic with **`procsd update --or-create`** instead of `procsd create --or-update` for deployment. Because:

* `update` by default calling `restart` even if services wasn't changed so we don't need to provide `--or-restart` flag or use additional  `after  :finished,    'procsd:restart'` for capistrano
* `update --or-create` in general makes more sense for deployment than `create --or-update` because `create` needs to call only for a first time, where `update` required for any second and more deploy (except for a first deploy)

**Pull request closed without merging**